### PR TITLE
[Refactor] Set request state final variables

### PR DESCRIPTION
**Pull request by torkelbe**
### Changes
* Set final variables for request state in Headers constructor
    - `clientRequestUrlWithoutLangCode`
    - `shouldRedirectToDefaultLang`
    - `isValidPath`
* Use UrlLanguagePatternHandler to check if request URL matches sitePrefixPath
* Ensure that `removeLang` accepts being called with empty language argument

### CODE REVIEW

File: src/main/java/com/github/wovnio/wovnjava/Interceptor.java [hunk 1]
```diff
@@ -16,7 +16,7 @@
     }
 
     String translate(String body) {
-        String lang = headers.getRequestLang();
+        String lang = headers.requestLang;
```
Comment (a-bougouin) [inline src/main/java/com/github/wovnio/wovnjava/Interceptor.java]: I am not a huge fan of having `requestLang` public. It breaks uniform access, now we use either public attributes or getters. I prefer using getters for consistency.

File: src/main/java/com/github/wovnio/wovnjava/PathUrlLanguagePatternHandler.java [hunk 2]
```diff
@@ -26,6 +30,10 @@ String insertLang(String url, String lang) {
         return "site.com/en/path";
     }
 
+    boolean isMatchSitePrefixPath(String url) {
```
Comment (a-bougouin) [inline src/main/java/com/github/wovnio/wovnjava/PathUrlLanguagePatternHandler.java]: I think it should be named `isMatchingSitePrefixPath` or `matchesSitePrefixPath`.

File: src/main/java/com/github/wovnio/wovnjava/UrlLanguagePatternHandler.java [hunk 3]
```diff
@@ -10,6 +10,8 @@
 
     abstract String insertLang(String url, String lang);
 
+    abstract boolean isMatchSitePrefixPath(String url);
```
Comment (a-bougouin) [inline src/main/java/com/github/wovnio/wovnjava/UrlLanguagePatternHandler.java]: I would not make it abstract and implement default returning `true`. That would avoid having to think about it when implementing URL handler for non-path URL components. Only concerned handlers would override it.

Both solution are fine. I prefer the one I mentioned so I wanted to bring that up.

Comment (torkelbe) [reply src/main/java/com/github/wovnio/wovnjava/UrlLanguagePatternHandler.java]: That's nicer

File: src/main/java/com/github/wovnio/wovnjava/PathUrlLanguagePatternHandler.java (hunk 2 above)
Comment (torkelbe) [reply src/main/java/com/github/wovnio/wovnjava/PathUrlLanguagePatternHandler.java]: Now called `isMatchingSitePrefixPath`

File: src/main/java/com/github/wovnio/wovnjava/Interceptor.java (hunk 1 above)
Comment (torkelbe) [reply src/main/java/com/github/wovnio/wovnjava/Interceptor.java]: Headers currently has almost no getters, and fields are simply accessed directly. Consistency points to not using getters, if anything. That said, the old fields that are accessed directly will likely be removed soon.

I think I liked accessing the fields as if Headers is a struct, essentially. But I'm fine with using getters as well.

**Pull request merged**